### PR TITLE
Fix unknown credits crash

### PR DESCRIPTION
**Pull request by aaustin**
We have reports of a rare but unreproducible crash in `loadRewards`. The crash report indicated that there was a `Fatal Exception: NSInvalidArgumentException` on line 51. This pull request adds extensive type checking to ensure arguments are valid.

@agrimn @derrickstaten

### CODE REVIEW

**Comment by derrickstaten** [comment]
this looks fine to me. pending testing, 👍

File: Branch-SDK/Branch-SDK/Requests/BranchLoadRewardsRequest.m [hunk 1]
```diff
@@ -46,18 +46,26 @@ - (void)processResponse:(BNCServerResponse *)response error:(NSError *)error {
...
     NSArray *storedKeys = [currentCreditDictionary allKeys];
 
-    if ([responseKeys count]) {
+    if ([responseKeys count] && ([response.data isKindOfClass:[NSDictionary class]] || [response.data isKindOfClass:[NSMutableDictionary class]])) {
+        
         for (NSString *key in response.data) {
-             NSInteger credits = [response.data[key] integerValue];
+            if (![key isKindOfClass:[NSString class]]) { continue; }
+            
+            NSInteger credits = [preferenceHelper getCreditCountForBucket:key];
```
Comment (derrickstaten) [inline Branch-SDK/Branch-SDK/Requests/BranchLoadRewardsRequest.m]: why did the `credits` variable change from being derived from the response to being pulled from preference helper?

Comment (derrickstaten) [inline Branch-SDK/Branch-SDK/Requests/BranchLoadRewardsRequest.m]: other than this, 👍

Comment (aaustin) [inline Branch-SDK/Branch-SDK/Requests/BranchLoadRewardsRequest.m]: that's harmless because it gets set in the pref helper down below

Comment (derrickstaten) [inline Branch-SDK/Branch-SDK/Requests/BranchLoadRewardsRequest.m]: got it. i guess this has no impact on `hasUpdated` which is all i cared about. certainly an edge case.

**Comment by derrickstaten** [comment]
👍

**Pull request merged**